app/src/main/java/com/weatherxm/util/Weather.kt

```diff
@@ -106,15 +106,13 @@ object Weather : KoinComponent {
             return "$EMPTY_VALUE$unit"
         }
 
-        return if (ignoreConversion) {
-            if (decimals == 0) {
-                "${roundToInt(value)}$unit"
-            } else {
-                "${roundToDecimals(value)}$unit"
-            }
+        val formattedValue = if (ignoreConversion) {
+            formatNumber(value, decimals)
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/WeatherXM/wxm-android/pull/342#issuecomment-2391143597>#b921d3</a><span> - 10/03/2024, 11:34 am</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The formatNumber function is used without considering the possibility of null return values, which could lead to potential null pointer exceptions.
</div>
<div id="fix">
<b>Fix:</b> Use the Elvis operator (?:) to provide a default value in case formatNumber returns null. This ensures that a non-null value is always used.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -109,7 +109,7 @@ object Weather : KoinComponent {
      }

      val formattedValue = if (ignoreConversion) {
 -        formatNumber(value, decimals)
 +        formatNumber(value, decimals) ?: EMPTY_VALUE
      } else {
 -        formatNumber(convertTemp(value, decimals), decimals)
 +        formatNumber(convertTemp(value, decimals), decimals) ?: EMPTY_VALUE
      }
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged